AwsEncryptionSDK/runtimes/net/TestVectorsNative/TestVectors/TestVectors.cs

```diff
@@ -133,144 +135,195 @@ public TestVectorDecryptTests(ITestOutputHelper testLogging)
+                    // Use Reflection to get the common Obj field
+                    Type opaqueType = e.GetType();
+                    FieldInfo objFieldInfo = opaqueType.GetField("obj");
+                    object obj = objFieldInfo?.GetValue(e);
+                    switch (obj)
+                    {
+                        case null when e.Message.Equals(OPAQUE_ERROR_NULL_OBJ_MSG):
+                            testLogging.WriteLine($"OpaqueError Logging: Obj was null. Error Type is {opaqueType}");
+                            exceptionHolder = e;
+                            break;
+                        case Exception nestedException:
+                            testLogging.WriteLine($"OpaqueError Logging: Obj is an Exception. " +
+                                                  $"Error Type is {opaqueType}.\n" +
+                                                  $"Nested Exception is {nestedException.GetType()}.\n" +
+                                                  $"Nested Exceptions message is: \n\t{nestedException.Message}\n"
+                                // + $"Nested Exceptions StackTrace is: \n\t{nestedException.StackTrace}"
+                            );
+                            exceptionHolder = nestedException;
+                            break;
+                        default:
+                            testLogging.WriteLine($"OpaqueError Logging: Obj is an arbitrary object. " +
+                                                  $"Error Type is {opaqueType}. " +
+                                                  $"Type of Obj is {obj!.GetType()}.");
+                            exceptionHolder = e;
+                            break;
+                    }
+
+                    break;
                 }
+                default:
+                    exceptionHasBeenCaught = true;
+                    exceptionHolder = e;
+                    testLogging.WriteLine($"Unexpected Exception: {e}");
+                    testLogging.WriteLine($"Unexpected Exception: Error Type is {e.GetType()}. " +
+                                          $"Exception's message is: {e.Message}");
+                    break;
+            }
+            
+            if (expectedPlaintext != null && exceptionHasBeenCaught)
+            {
+                testLogging.WriteLine("Decrypt Failed, and should not have!");
             }
+            
             if (exceptionHolder != null && expectedPlaintext != null)
             {
-                // Should succeed but did not, throw exception
+                if (exceptionHolder is System.Security.Cryptography.CryptographicException
+                    or Org.BouncyCastle.Crypto.InvalidCipherTextException)
+                {
+                    // Should succeed but did not, throw exception
+                    throw new TestVectorShouldHaveFailedException(exceptionHolder.Message);
+                }
+
                 throw exceptionHolder;
             }
         }
```

Comment:
I am reading:

(309) if plaintext is expected and an exception has been caught: log

(314) if plaintext is expected and an exception has been stored: throw

I think that is the same behavior I had...
But I have a hard time reasoning if it is correct.